jaxrs/src/main/java/org/jboss/as/jaxrs/deployment/JaxrsIntegrationProcessor.java

```diff
@@ -458,5 +469,145 @@ public static void setContextParameter(JBossWebMetaData webdata, String name, St
+            setContextParameter(webdata, JaxrsConstants.RESTEASY_PREFER_JACKSON_OVER_JSONB, modelNode.asString());
+        }
+        if (isTransmittable(JaxrsAttribute.RESTEASY_PROVIDERS, modelNode = config.getResteasyProviders())) {
+            setContextParameter(webdata, JaxrsConstants.RESTEASY_PROVIDERS, convertListToString(modelNode));
+        }
+        if (isTransmittable(JaxrsAttribute.RESTEASY_RFC7232_PRECONDITIONS, modelNode = config.isResteasyRFC7232Preconditions())) {
+            setContextParameter(webdata, JaxrsConstants.RESTEASY_RFC7232_PRECONDITIONS, modelNode.asString());
+        }
+        if (isTransmittable(JaxrsAttribute.RESTEASY_ROLE_BASED_SECURITY, modelNode = config.isResteasyRoleBasedSecurity())) {
+            setContextParameter(webdata, JaxrsConstants.RESTEASY_ROLE_BASED_SECURITY, modelNode.asString());
+        }
+        if (isTransmittable(JaxrsAttribute.RESTEASY_SECURE_RANDOM_MAX_USE, modelNode = config.getResteasySecureRandomMaxUse())) {
+            setContextParameter(webdata, JaxrsConstants.RESTEASY_SECURE_RANDOM_MAX_USE, modelNode.asString());
+        }
+        if (isTransmittable(JaxrsAttribute.RESTEASY_USE_BUILTIN_PROVIDERS, modelNode = config.isResteasyUseBuiltinProviders())) {
+            setContextParameter(webdata, JaxrsConstants.RESTEASY_USE_BUILTIN_PROVIDERS, modelNode.asString());
+        }
+        if (isTransmittable(JaxrsAttribute.RESTEASY_USE_CONTAINER_FORM_PARAMS, modelNode = config.isResteasyUseContainerFormParams())) {
+            setContextParameter(webdata, JaxrsConstants.RESTEASY_USE_CONTAINER_FORM_PARAMS, modelNode.asString());
+        }
+        if (isTransmittable(JaxrsAttribute.RESTEASY_WIDER_REQUEST_MATCHING, modelNode = config.isResteasyWiderRequestMatching())) {
+            setContextParameter(webdata, JaxrsConstants.RESTEASY_WIDER_REQUEST_MATCHING, modelNode.asString());
+        }
+    }
+
+    /**
+     * Send value to RESTEasy only if it's not null, empty string, or the default value.
+     */
+    private boolean isTransmittable(AttributeDefinition attribute, ModelNode modelNode) {
+        if (modelNode == null) {
+            return false;
+        }
+        String value = modelNode.asString();
+        if ("".equals(value.trim())) {
+            return false;
+        }
+        return !value.equals(attribute.getDefaultValue());
+    }
+
+    /**
+     * List attributes can be reset to white space, but RESTEasy's ConfigurationBootstrap doesn't handle
+     * empty maps appropriately at present.
+     */
+    private boolean isSubstantiveList(ModelNode modelNode) {
+        if (modelNode == null) {
+            return false;
+        }
+        String value = modelNode.asString();
+        return value != null && !"".equals(value.trim());
+    }
+
+    private String convertListToString(ModelNode modelNode) {
+        StringBuffer sb = new StringBuffer();
+        String[] ss = strip(modelNode.asString().trim(), '[', ']').split(",");
```

Comment:
Rather than do this, it is better to do
```
for (ModelNode value : modelNode.asList()) {
    // TODO Add comma depending on value of first

    sb.append(value.asString());
}

Reply:
Ah, ok.

Reply:
I realize that I had some old code in JaxrsParamHandler that was saving resolved ModelNodes as strings, so JaxrsIntegrationProcessor was getting strings instead of lists and objects. I fixed that, and now your simpler code works.